zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/tenant/DeleteTenantAuthorizationTest.java

```diff
@@ -0,0 +1,136 @@
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.tenant;
+
+import static io.camunda.zeebe.protocol.record.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.security.configuration.ConfiguredUser;
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.TenantIntent;
+import io.camunda.zeebe.protocol.record.value.AuthorizationOwnerType;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.EntityType;
+import io.camunda.zeebe.protocol.record.value.PermissionType;
+import io.camunda.zeebe.protocol.record.value.TenantRecordValue;
+import io.camunda.zeebe.protocol.record.value.UserRecordValue;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+
+public class DeleteTenantAuthorizationTest {
+
+  private static final ConfiguredUser DEFAULT_USER =
+      new ConfiguredUser(
+          UUID.randomUUID().toString(),
+          UUID.randomUUID().toString(),
+          UUID.randomUUID().toString(),
+          UUID.randomUUID().toString());
+
+  @Rule
+  public final EngineRule engine =
+      EngineRule.singlePartition()
+          .withIdentitySetup()
+          .withSecurityConfig(
+              cfg -> {
+                cfg.getAuthorizations().setEnabled(true);
+                cfg.getInitialization().setUsers(List.of(DEFAULT_USER));
+                cfg.getInitialization()
+                    .getDefaultRoles()
+                    .put("admin", Map.of("users", List.of(DEFAULT_USER.getUsername())));
+                cfg.getMultiTenancy().setEnabled(true);
+              });
+
+  @Rule public final TestWatcher recordingExporterTestWatcher = new RecordingExporterTestWatcher();
+
+  @Test
+  public void shouldBeAuthorizedToDeleteAssignedTenantWithPermissions() {
+    // given
+    final var user = createUser();
+    final var tenantId = UUID.randomUUID().toString();
+    createTenant(tenantId);
+    addPermissionsToUser(user, AuthorizationResourceType.TENANT, PermissionType.DELETE, tenantId);
+    assignUserToTenant(user.getUsername(), tenantId);
+
+    // when
+    engine.tenant().deleteTenant(tenantId).delete(user.getUsername());
+
+    // then
+    assertThat(
+            RecordingExporter.tenantRecords()
+                .withTenantId(tenantId)
+                .withIntent(TenantIntent.DELETED)
+                .count())
+        .isOne();
```

Comment:
🔧 We should change this assertion so it breaks out of the 5 second wait. Since we are collecting a count and checking it's 1, how about using exists() and true?